tests/model/SiteTreeTest.php

```diff
@@ -108,6 +109,47 @@ public function testURLGeneration()
     }
 
     /**
+     * Check if reserved URL's are properly appended with a number at top level
+     */
+    public function testDisallowedURLGeneration()
+    {
+        // Expected URL defaults to '-2'
+        $expectedURLs = [
```

Comment:
you might want to consider the use of data providers (https://phpunit.de/manual/5.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers)